cpp/src/copying/contiguous_split.cu

```diff
@@ -225,6 +226,33 @@ __global__ void copy_partition(int num_src_bufs,
               buf_info[buf_index].bit_shift);
 }
 
+/**
+ * @brief Add metadata information to the metadata buffer.
+ *
+ * @param metadata Metadata to add information to
+ * @param type Column type
+ * @param size Column size
+ * @param null_count Column null count
+ * @param data_offset Offset into contiguous data buffer, or -1 if pointer should deserialize to
+ * null
+ * @param null_mask_offset Offset into contiguous data buffer, or -1 if pointer should deserialize
+ * to null
+ * @param num_children # of children
+ */
+void add_column_metadata(std::vector<uint8_t>& metadata,
+                         data_type type,
+                         size_type size,
+                         size_type null_count,
+                         int64_t data_offset,
+                         int64_t null_mask_offset,
+                         size_type num_children)
+{
+  detail::serialized_column column_metadata{
+    type, size, null_count, data_offset, null_mask_offset, num_children};
+  auto const bytes = reinterpret_cast<uint8_t const*>(&column_metadata);
+  std::copy(bytes, bytes + sizeof(detail::serialized_column), std::back_inserter(metadata));
```

Comment:
Avoid the copy and just construct in place. 

```suggestion
 new (metadata.data()) detail::serialized_column{type, size, null_count, data_offset, null_mask_offset, num_children};
```

Reply:
Couple of issues here.  .data() just returns the head of the vector, not the current element.  And the metadata vector isn't pre-sized at this point.

Reply:
Hm, okay, I misunderstood what was originally happening. So I see a couple of alternatives.

You could build up the metadata vector as a `vector<detail::serialize_column>` and then at the end copy it in bulk to a`std::vector<char>`. But that still requires an unnecessary copy.

This is a little messier than what I had before, but still isn't too bad:
```
column_metadata.reserve(column_metadata.capacity() + sizeof(detail::serialize_column));
auto p = std::addressof(column_metadata.back());
column_metadata.push_back(sizeof(detail::serialized_column));
new (p) detail::serialized_column{...};
```

Reply:
I don't think this will work.  You can't call .back() on an empty vector (even though it's got reserved space).  Also push_back() is going to try and push a byte, not a # of bytes == sizeof(detail::serialized_column).  

Ultimately, I don't think there's a way to trick std::vector into _not_ initializing the underlying bytes before you try the placement new.